src/bridges/mean/MeanBridge.sol

```diff
@@ -31,13 +35,37 @@ contract MeanBridge is BridgeBase, Ownable2Step {
      * @param _owner The account that will own the bridge
      * @param _rollupProcessor Address of rollup processor
      */
-    constructor(IDCAHub _hub, address _owner, address _rollupProcessor) BridgeBase(_rollupProcessor) {
+    constructor(IDCAHub _hub, ITransformerRegistry _transformerRegistry, address _owner, address _rollupProcessor) BridgeBase(_rollupProcessor) {
         _transferOwnership(_owner);
         DCA_HUB = _hub;
+        TRANSFORMER_REGISTRY = _transformerRegistry;
         THIS_ADDRESS = address(this);
     }
 
     // Note: we need to be able to receive ETH to deposit as WETH
     receive() external payable {}
 
+    /**
+     * @notice Set all the necessary approvals for the DCA hub, the transformer registry and the rollup processor
+     * @param _tokens - The tokens to approve
+     */
+    function maxApprove(IERC20[] calldata _tokens) external {
+        for (uint i; i < _tokens.length; ) {
+            IERC20 _token = _tokens[i];
+            _maxApprove(_token, address(DCA_HUB));
+            _maxApprove(_token, address(TRANSFORMER_REGISTRY));
+            _maxApprove(_token, address(ROLLUP_PROCESSOR));
+            unchecked {
+                ++i;
+            }
+        }
+    }
```

Comment:
Not sure I am a fan / understand why is this fully permissionless? Seems like an unnecessary risk vector.

Reply:
Why would it be risky? We are just approving tokens. All other bridges do this in a permissionless fashion also

Reply:
I was going to say that we need to trust that the DCA HUB and TRANSFORMER REGISTRY are not vulnerable (which... we trust, just airing on the side of "less-privilege" architecture), but either way we need to approve it to make it cheaper. Whatever you feel here it's ok. I just don't see big down-side on making it permissioned.

Reply:
I don't think it's a big down-side, but since the bridge doesn't hold any funds, and we trust our own contracts, I think it's easier to simply make it permissionless